### PR TITLE
Ensure the Baremetal CRD is installed before metal3 is deployed

### PR DESCRIPTION
**Pull request by sadasu**
Moving installation of Baremetal CRD before Metal3 deployment happens.
Also, adding the baremetal CRD to kustomization.yaml to keep it up-to-date.
Fixes issue https://github.com/openshift/machine-api-operator/issues/397.

### CODE REVIEW

File: kustomization.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ resources:
 - install/0000_30_machine-api-operator_10_service.yaml
```
[the remark this answers is not in the document]
Comment (ingvagabund) [reply kustomization.yaml]: Correct. This is only for testing, not for production.

[the remark this answers is not in the document]
Comment (sadasu) [reply kustomization.yaml]: Sorry. Saw that this baremetal CRD file was the only one missing from this and didn't realize that kustomization.yaml was not being used anymore. Should we remove  the file entirely from the repo?

[the remark this answers is not in the document]
Comment (ingvagabund) [reply kustomization.yaml]: `kustomization.yaml` can be still used to deploy mao over minikube. It's primarily for development.

**Comment by sadasu** [comment]
@bison @ingvagabund  can you please provide your reviews?

**Comment by enxebre** [comment]
can you please update the  description? the `kustomization.yaml` file is just a helper for devs atm, the rest of the content in the description does not applies to this PR

**Comment by sadasu** [comment]
@michaelgugino would you be able to review this?

**Pull request merged**